clients/cobol-lsp-vscode-extension/src/CopybooksDownloader.ts

```diff
@@ -14,81 +14,64 @@
             this.downloadDependencies(dep, message);
         });
     }
     /**
      * @param copybooks array of copybooks names to download
      */
-    public async downloadDependencies(uri: vscode.Uri, message: string = ""): Promise<void> {
+    async downloadCopybooks(copybooks: string[], programName: string): Promise<void> {
+        // TODO do it right
+        const profile: string = await this.profileService.getProfile(programName);
+        if (!profile) {
+            return;
+        }
+        copybooks.forEach(copybook => this.queue.push(copybook, profile));
+    }
+
+    public async downloadDependencies(depFileUri: vscode.Uri, message: string = ""): Promise<void> {
         if (!checkWorkspace()) {
             return;
         }
-        const profile: string = await this.profileService.getProfile(uri);
+        const depDesc: DependenciesDesc = loadDepFile(depFileUri);
+        const profile: string = await this.profileService.getProfile(depDesc.programName);
         if (!profile) {
             return;
         }
 
-        const missingCopybooks: string[] = await this.listMissingCopybooks(uri, profile);
+        const missingCopybooks: string[] = await this.listMissingCopybooks(depDesc.copybooks, profile);
 
         if (!message.length) {
             missingCopybooks.forEach(copybook => this.queue.push(copybook, profile));
         } else if (missingCopybooks.length > 0) {
-            // TODO: refactor: move to external class
-            const downloadCopybookAction = "Download Copybooks";
-            const actionDatasets = "Edit Datasets";
-            const actionProfile = "Change zowe profile";
-            const actions = [];
-            if ((await this.listPathDatasets()).length > 0) {
-                actions.push(downloadCopybookAction);
-            }
-            if (message !== "Configuration was updated") {
-                actions.push(actionDatasets);
-                actions.push(actionProfile);
-            }
-            const action: string = await vscode.window.showInformationMessage(
-                message,
-                ...actions);
-            if (action === downloadCopybookAction) {
-                missingCopybooks.forEach(copybook => this.queue.push(copybook, profile));
-            }
-            if (action === actionDatasets) {
-                vscode.commands.executeCommand("workbench.action.openSettings",
-                    "broadcom-cobol-lsp.cpy-manager.paths");
-            }
-            if (action === actionProfile) {
-                vscode.commands.executeCommand("workbench.action.openSettings",
-                    "broadcom-cobol-lsp.cpy-manager.profiles");
-            }
+            this.resolver.fixMissingDownloads(message, missingCopybooks, profile, {
+                hasPaths: (await this.listPathDatasets()).length > 0,
+            });
         }
 
     }
 
     // tslint:disable-next-line: cognitive-complexity
     public async start() {
```

Comment:
Ideally we should take care of the cognitive complexity.
If there is no space at moment to fix that please file an issue on GH to tackle it later on with a specific fix.

Reply:
It turns out it's not so complex after all )

Reply:
Well.. when I removed the tslint I got the warning about complexity 28 lines instead of 15. 
But I'm fine with that.

Reply:
Hm.. didn't get this warning on my site. Sorry, I'll double-check.

Reply:
![image](https://user-images.githubusercontent.com/48674737/76207112-7b365580-61fd-11ea-98fb-6c1cee521af2.png)

Reply:
Btw I think we can plan a small refactor later on